functions/Get-DbaBuildReference.ps1

```diff
@@ -58,16 +73,38 @@ function Get-DbaBuildReference {
 
     #>
     [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSUseShouldProcessForStateChangingFunctions", "")]
-    [CmdletBinding()]
+    [CmdletBinding(DefaultParameterSetName = 'Build')]
```

Comment:
we've tried to move away from parametersets because they error really weirdly and confuse people. i know we have some commands that still use them but they're generally the older ones in need of a refactor.

can you change these into stop functions? if it's too much work/ablocker, we can do it later.

Reply:
This one I'm slightly confused about. I love ParameterSets and if some of them confuse people, then the developer did something very wrong. Get-Help with defined parameterSets is much more nice, providing parameters grouped by usage. And they error weirdly when people are not providing proper parameters, not because parameterSets are evil. 
In this particular scenario, I want to be able to call function in four different ways. I don't see a benefit in mashing all the parameters together and leave users to figure it out, when we can define independent sets and have a nice and friendly way of confining users in appropriate boundaries.
But this is definitely not the hill where I'm ready to make a stand. If you say stop-fucntions, that's the way to go.

Reply:
> then the developer did something very wrong. 

Perhaps, but it's the way I naturally work. I didn't realize what the different blocks were in Get-Help for years, and always had to google the weird generic "parameter could not be resolved" error, and I spent about an hour before realizing I used two commands from conflicting parametersets. I would have appreciated if I was just told "You cannot specify Y when using X". But PS hasn't given us that option so we all decided a while back to give clarity to our users by telling them in human words that they can't use X when using Y. I wish parametersets would have been the solution, but they're just not user-friendly.